sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/RedundantCast.CSharp8.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Tests.Diagnostics
+{
+    public class CastOnNullable
+    {
+        public static IEnumerable<string> UsefulCast()
+        {
+            var nullableStrings = new string?[] { "one", "two", null, "three" };
```

Comment:
Using C# 6 with regular strings produces the same result - skipping `null` value. I think it should be FP for `string` as well (when it's FP for `string?`).
```C#
var normalStrings = new string[] { "one", "two", null, "three" };
```

Reply:
that's a very good point!